kalite/coachreports/static/js/coachreports/coach_reports/views.js

```diff
@@ -0,0 +1,343 @@
+        });
+    }
+});
+
+
+var detailsPanelBodyView = BaseView.extend({
+    
+    template: HB.template("coach_nav/detailspanelbody"),
+    
+    initialize: function (options) {
+        _.bindAll(this);
+        this.render();
+    },
+    
+    render: function() {
+        this.$el.html(this.template({
+            collection: this.collection.to_objects()
+        }));
+    }
+});
+
+var TabularReportView = BaseView.extend({
+
+    template: HB.template("tabular_reports/tabular-view"),
+
+    initialize: function() {
+        _.bindAll(this);
+        this.set_data_model();
+        this.listenTo(this.model, "change", this.set_data_model);
+    },
+
+    render: function() {
+        this.$el.html(this.template({
+            contents: this.contents.toJSON(),
+            learners: this.contents.length
+        }));
+        var row_views = [];
+        for (var i = 0; i < this.learners.length; i++) {
+            var row_view = this.add_subview(TabularReportRowView, {model: this.learners.at(i), contents: this.contents})
+            row_views.push(row_view);
+            this.listenTo(row_view, "detail_view", this.set_detail_view);
+        }
+        this.append_views(row_views, ".student-data");
+    },
+
+    set_data_model: function (){
+        var self = this;
+        this.data_model = new CoachReportModel({
+            facility: this.model.get("facility"),
+            group: this.model.get("group")
+        });
+        if (this.model.get("facility")) {
+            this.data_model.fetch().then(function() {
+                self.learners = new Backbone.Collection(self.data_model.get("learners"));
+                self.contents = new Backbone.Collection(self.data_model.get("contents"));
+                for (var i = 0; i < self.learners.length; i++) {
+                    self.learners.models[i].set("logs", _.object(
+                        _.map(_.filter(self.data_model.get("logs"), function(log) {
+                            return log.user === self.learners.models[i].get("pk");
+                        }), function(item) {
+                            return [item.exercise_id || item.video_id || item.content_id, item];
+                        })));
+                }
+                self.render();
+            });
+        }
+    },
+
+    set_detail_view: function(detail_view) {
+        if (this.detail_view) {
+            this.detail_view.remove();
+        }
+        if (detail_view) {
+            this.detail_view = detail_view;
+        }
+    }
+
+});
+
+var TabularReportRowView = BaseView.extend({
+
+    template: HB.template("tabular_reports/tabular-view-row"),
+
+    tagName: 'tr',
+
+    initialize: function(options) {
+        _.bindAll(this);
+
+        this.contents = options.contents;
+        this.render();
+    },
+
+    render: function() {
+        this.$el.html(this.template(this.model.attributes));
+
+        var cell_views = [];
+        for (var i = 0; i < this.contents.length; i++) {
```

Comment:
Just a matter of taste, but you could use (_.each)[http://underscorejs.org/#each] here, I think it's even a Collection method.